ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -133,11 +132,18 @@ class IndexSegment implements Iterable<IndexEntry> {
     this.factory = factory;
     this.metrics = metrics;
     this.time = time;
+    // When PersistentIndex tries to add one IndexEntry and it decides to create a new IndexSegment,
+    // it provides that new IndexEntry's valueSize and entrySize for the IndexSegment.
+    // Inside one IndexSegment, all the index entries have the same valueSize.
```

Comment:
// Inside one IndexSegment, all the index entries have the same valueSize.

i don't get this, i thought all entries in an index-segment are sorted by blob id. 
Is this an additional constraint that all entries must have same valueSize ?
this might disrupt the sorted order...

Reply:
Again, valueSize is the size of the IndexValue.
Why will valueSize impact the sorting?
Basically all the Index Value in the same IndexSegment are with the same PersistentIndex.VERSION_X.